cli_tools/common/utils/param/populator.go

```diff
@@ -45,15 +45,15 @@ type populator struct {
 	scratchBucketClient domain.ScratchBucketCreatorInterface
 }
 
-func (p populator) PopulateMissingParameters(project *string, zone *string, region *string,
-	scratchBucketGcsPath *string, file string, storageLocation *string) error {
+func (p populator) PopulateMissingParameters(project *string, clientID string, zone *string,
+	region *string, scratchBucketGcsPath *string, file string, storageLocation *string) error {
 
 	if err := PopulateProjectIfMissing(p.metadataClient, project); err != nil {
 		return err
 	}
 
 	scratchBucketRegion, err := populateScratchBucketGcsPath(scratchBucketGcsPath, *zone, p.metadataClient,
-		p.scratchBucketClient, file, project, p.storageClient)
+		p.scratchBucketClient, file, project, p.storageClient, clientID == "gcloud")
```

Comment:
We should consider cleaning up for empty client IDs as well. Those are either old gcloud versions or API calls that are not following the pattern in our examples AND are importing files that they've copied to a scratch bucket. I think both of these criteria being true at the same is unlikely, though not impossible. I'm not 100% sure which approach is better here TBH.

Reply:
Good news: we don't allow empty client IDs, either currently or historically:

* Current: https://github.com/GoogleCloudPlatform/compute-image-tools/blob/master/cli_tools/gce_vm_image_import/importer/args_test.go#L75
* Previous architecture: https://github.com/GoogleCloudPlatform/compute-image-tools/blob/v20190523/cli_tools/gce_vm_image_import/main.go#L104